rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize.cpp

```diff
@@ -54,10 +54,114 @@ void ResizeNode::create_node()
     vx_status status;
     if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
         THROW("Adding the resize (vxExtrppNode_ResizebatchPD) node failed: "+ TOSTR(status))
+}
 
+void ResizeNode::update_node() {
+    std::vector<uint32_t> src_h_dims, src_w_dims;
+    src_w_dims = _inputs[0]->info().get_roi_width_vec();
+    src_h_dims = _inputs[0]->info().get_roi_height_vec();
+    for (unsigned i = 0; i < _batch_size; i++) {
+        _src_roi_size[0] = src_w_dims[i];
+        _src_roi_size[1] = src_h_dims[i];
+        _dst_roi_size[0] = _dest_width;
+        _dst_roi_size[1] = _dest_height;
+        adjust_out_roi_size();
+        _dst_roi_size[0] = std::min(_dst_roi_size[0], _outputs[0]->info().width());
+        _dst_roi_size[1] = std::min(_dst_roi_size[1], _outputs[0]->info().height_single());
+        _dst_roi_width_vec.push_back(_dst_roi_size[0]);
+        _dst_roi_height_vec.push_back(_dst_roi_size[1]);
+    }
+    vx_status width_status, height_status;
+    width_status = vxCopyArrayRange((vx_array)_dst_roi_width, 0, _batch_size, sizeof(vx_uint32), _dst_roi_width_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    height_status = vxCopyArrayRange((vx_array)_dst_roi_height, 0, _batch_size, sizeof(vx_uint32), _dst_roi_height_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    if(width_status != 0 || height_status != 0)
+        WRN("ERROR: vxCopyArrayRange _dst_roi_width or _dst_roi_height failed " + TOSTR(width_status) + "  " + TOSTR(height_status));
+    _outputs[0]->update_image_roi(_dst_roi_width_vec, _dst_roi_height_vec);
+    _dst_roi_width_vec.clear();
+    _dst_roi_height_vec.clear();
 }
 
-void ResizeNode::update_node()
-{
-  
-}
\ No newline at end of file
+void ResizeNode::init(unsigned dest_width, unsigned dest_height, RocalResizeScalingMode scaling_mode,
+                      std::vector<unsigned> max_size, RocalResizeInterpolationType interpolation_type) {
+    _scaling_mode = scaling_mode;
+    _dest_width = dest_width;
+    _dest_height = dest_height;
+    _interpolation_type = (int)interpolation_type;
+    _src_roi_size.resize(2);
+    _dst_roi_size.resize(2);
+    _max_roi_size = max_size;
+}
+
+void ResizeNode::adjust_out_roi_size() {
+    const unsigned dim = 2; // Currently supports only 2D images
+    std::vector<double> scale(dim, 1);
```

Comment:
do we need to use double for scale instead of float?

Reply:
In future if we fuse resize with ROI similar to DALI API we will have src width and height values (in float), using double we were able to resolve the rounding off errors